sopel/modules/url.py

```diff
@@ -153,6 +153,96 @@ def shutdown(bot):
             pass
 
 
+@plugin.command('urlban', 'urlpban')
+@plugin.example('.urlpban example\\.com/\\w+', user_help=True)
+@plugin.example('.urlban example.com/path', user_help=True)
+@plugin.output_prefix('[url] ')
+def url_ban(bot, trigger):
+    """Ban a URL from auto title.
+
+    Use ``urlpban`` to ban a pattern instead of a URL.
+    """
+    url = trigger.group(2)
+
+    if not url:
+        bot.reply('This command requires an URL to ban.')
+        return
+
+    if trigger.group(1) == 'urlpban':
+        # validate regex pattern
+        try:
+            re.compile(url)
+        except re.error as err:
+            bot.reply('Invalid regex pattern: %s' % err)
+            return
+    else:
+        # escape the URL to ensure a valid pattern
+        url = re.escape(url)
+
+    patterns = bot.settings.url.exclude or []
```

Comment:
No need to fall back on an empty list; `ListAttribute` already presents as empty list if not set.

```suggestion
    patterns = bot.settings.url.exclude
```